### PR TITLE
Adding EcoData Dataset scripts

### PR DESCRIPTION
**Pull request by zhangcandrew**
These need to be further tested, but my code for try_install_all isn't working properly I don't think. If you find the time @henrykironde to look through it, that'd be great! I'll be looking at it when I can to update it. Just thought it was about time this PR went up. 

When try_install_all works and the scripts have been fully tested, I will remove try_install_all.py from the PR. I just have it up there for people to look at and possibly make suggestions.

### CODE REVIEW

**Comment by henrykironde** [comment]
Excited to test these

**Comment by zhangcandrew** [comment]
Scripts have been tested, all tests have passed for me. Everything looks good. Let me know how it works for you guys.

File: scripts/Barnes2008.json [hunk 1]
```diff
@@ -0,0 +1,24 @@
+{
+    "citation": "C. Barnes, D. M. Bethea, R. D. Brodeur, J. Spitz, V. Ridoux, C. Pusineri, B. C. Chase, M. E. Hunsicker, F. Juanes, A. Kellermann, J. Lancaster, F. M\u00e9nard, F.-X. Bard, P. Munk, J. K. Pinnegar, F. S. Scharf, R. A. Rountree, K. I. Stergiou, C. Sassa, A. Sabates, and S. Jennings. 2008. Predator and prey body sizes in marine food webs. Ecology 89:881.",
```
Comment (henrykironde) [inline scripts/Barnes2008.json]: M\u00e9nard, >> Ménard

File: scripts/Rebecca2005.json [hunk 2]
```diff
@@ -0,0 +1,21 @@
+{
+    "citation": "Rebecca A. Bartel, Frederick F. Knowlton, and Charles Stoddart. 2005. Mammal abundance indices in the northern portion of the Great Basin, 1962\u20131993. Ecology 86:3130.",
```
Comment (henrykironde) [inline scripts/Rebecca2005.json]: 1962-1993. Ecology 86:3130.

File: scripts/Schmid2009.json [hunk 3]
```diff
@@ -0,0 +1,24 @@
+{
+    "citation": "Bernhard Schmid, Andrea B. Pfisterer, and Patricia Balvanera. 2009. Effects of biodiversity on ecosystem community, and population variables reported 1974\u20132004. Ecology 90:853.",
```
Comment (henrykironde) [inline scripts/Schmid2009.json]: 1974-2004. Ecology 90:853

File: scripts/Sinoquet2009.json [hunk 4]
```diff
@@ -0,0 +1,59 @@
+{
+    "citation": "Herv\u00e9 Sinoquet, Sylvain Pincebourde, Boris Adam, Nicolas Don\u00e8s, Jessada Phattaralerphong, Didier Combes, St\u00e9phane Ploquin, Krissada Sangsing, Poonpipope Kasemsap, Sornprach Thanisawanyangkura, G\u00e9raldine Groussier-Bout, and J\u00e9r\u00f4me Casas. 2009. 3-D maps of tree canopy geometries at leaf scale. Ecology 90:283",
```
Comment (henrykironde) [inline scripts/Sinoquet2009.json]: `Hervé Sinoquet, Sylvain Pincebourde, Boris Adam, Nicolas Donès, Jessada Phattaralerphong, Didier Combes, Stéphane Ploquin, Krissada Sangsing, Poonpipope Kasemsap, Sornprach Thanisawanyangkura, Géraldine Groussier-Bout, and Jérôme Casas. 2009. 3-D maps of tree canopy geometries at leaf scale. Ecology 90:283`

File: scripts/paquette2007.json [hunk 5]
```diff
@@ -0,0 +1,49 @@
+{
+    "citation": "Alain Paquette, Etienne Lalibert\u00e9, Andr\u00e9 Bouchard, Sylvie de Blois, Pierre Legendre, and Jacques Brisson. 2007. Lac Croche understory vegetation data set (1998\u20132006). Ecology 88:3209.",
```
Comment (henrykironde) [inline scripts/paquette2007.json]: Alain Paquette, Etienne Laliberté, André Bouchard, Sylvie de Blois, Pierre Legendre, and Jacques Brisson. 2007. Lac Croche understory vegetation data set (1998-2006). Ecology 88:3209.

File: try_install_all.py [hunk 6]
```diff
@@ -11,10 +11,34 @@
...
+    subprocess.check_output(
+    'git fetch https://github.com/weecology/retriever.git master 2>/dev/null',
+    shell=True)
+
+    output = subprocess.check_output(
+                "git diff --name-only FETCH_HEAD ../scripts",
+                shell=True)
+
+    # get list of changed scripts
+    scripts_changed = to_string(output).splitlines()
```
Comment (henrykironde) [inline try_install_all.py]: os.path.basename(n)

**Comment by henrykironde** [comment]
When dealing with citation, the easy way to avoid errors is pasting the string in you browser's search field and copying it again from there. 

Some text editors if not well set will try to decode non-ascii characters and may fail if the character mapping is not obtaines hence returning the byte representation.

Also some scripts have these characters ``-, : ``,. Most of the time they are decoded wrongly. I usually test by asking retriever for the script citation and if errors occur,  I rewrite these characters``-, : ``,

File: scripts/Falcone2010.json [hunk 7]
```diff
@@ -0,0 +1,196 @@
+{
+    "citation": "James A. Falcone, Daren M. Carlisle, David M. Wolock, and Michael R. Meador. 2010. GAGES:A stream gage database for evaluating natural and altered flow conditions in the conterminous United States. Ecology 91:621.",
+    "description": "streamflow in ecosystems",
```
Comment (zhangcandrew) [inline scripts/Falcone2010.json]: I think it is "streamflow" if that's what you were asking.

**Comment by zhangcandrew** [comment]
Thanks for all that Henry! I made the changes. Working on renaming the files to be more descriptive now

**Comment by zhangcandrew** [comment]
These are the proposed new names I have for the scripts

Schmid2009.json   ->   scripts/biodiversity_response.json
	Armbruster2011.json   ->   scripts/bird_migration_data.json
	OliverButterfly2006.json   ->   scripts/butterfly_diversity_survey.json
	Murray2010.json   ->   scripts/chytr_disease_distr.json
	paquette2007.json   ->   scripts/croche_vegetation_data.json
	Menges2008.json   ->   scripts/dicerandra_frutescens.json
	Zander2011.json   ->   scripts/flensburg_food_web.json
Rebecca2005.json   ->   scripts/great_basin_mammal_abundance.json
	Petermacroalgal2006.json   ->   scripts/macroalgal_communities.json
	Rassweiler2008.json   ->   scripts/macrocystis_variation.json
	Petraitis2009.json   ->   scripts/marine_recruitment_data.json
	PaulaPlantTraits2009.json   ->   scripts/mediter_basin_plant_traits.json
	Travers2009.json   ->   scripts/n.greatplains-flowering-dates.json
	MulderNematode2011.json   ->   scripts/nematode_traits.json
	Raymond2011.json   ->   scripts/ocean_diet_data.json
	UlrichBodySizes.json   ->   scripts/predator_prey_body_ratio.json
	Cardinale2009.json   ->   scripts/species_exctinction_rates.json
	Falcone2010.json   ->   scripts/streamflow_conditions.json
	Sinoquet2009.json   ->   scripts/tree_canopy_geometries.json
	Schwanzturtle2009.json   ->   scripts/turtle-offspring-nesting.json

**Comment by zhangcandrew** [comment]
Renamed the scripts. You can test them with the try_install_all.py file. The scripts worked fine for me. When this is ready to go in, just let me know and I'll remove the try_install_all.py file from the PR.

**Comment by zhangcandrew** [comment]
Had to move some of these scripts to figshare since esapubs seems to have stopped working. Some other scripts I still need to do: 

```
flensburg_food_web.json -> .py due to figshare zip file
mediter_basin_plant_traits.json -> might not be on figshare?
predator_prey_body_ratio.json -> not on figshare?
tree_canopy_geometries.json -> do later
streamflow_conditions.json -> do later
```

**Comment by ethanwhite** [comment]
If it still looks like there are missing datasets after another look around let me know which ones and I'll ask into what's going on with them with the ESA/Figshare folks.

**Comment by zhangcandrew** [comment]
Finishing up all sets. I found the other datasets, just working on the Flensburg python file now and then I'll remove the try_install_all.py from the PR, and this should be good.

**Comment by zhangcandrew** [comment]
The commits and everything got a little hairy on the branch I was working on, moved this to a new PR.

**Pull request closed without merging**